src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -75,17 +72,8 @@ public bool IsExperimentEnabled(string experimentName)
                     // since we use this for experimentation service as well
                 }
 
-                try
-                {
-                    return (bool)_isCachedFlightEnabledInfo.Invoke(_experimentationServiceOpt, new object[] { experimentName });
-                }
-                catch
-                {
-
-                }
+                return _experimentationService.IsCachedFlightEnabled(experimentName);
```

Comment:
I am not sure whether we can to throw here unless we know that it won't crash VS. otherwise, I would catch and fire NFW and return false so that we don't crash VS due to experiment failure.

Reply:
This is now a direct call instead of going through reflection. The latter had many more potential failure points. We can check if the implementation of this service ever throws; it may have its own internal error handling.

Reply:
This worries me: we'll be taking this for 16.2 GA so we may want to keep the catch in there to mitigate and then remove in 16.3.